### ISSUE TITLE
Enabling addons: default-storageclass, storage-provisioner

### ISSUE THREAD

**Issue by jwuman** [kubernetes/minikube#6709]
**The exact command to reproduce the issue**:
minikube start --cpus=2 --memory=4096 --vm-driver=kvm2

**The full output of the command that failed**:<details>

😄  minikube v1.7.3 on Arch 19.0.0
✨  Automatically selected the kvm2 driver. Other choices: virtualbox, none, docker (experimental), podman (experimental)
🔥  Creating kvm2 VM (CPUs=2, Memory=4096MB, Disk=20000MB) ...
🐳  Preparing Kubernetes v1.17.3 on Docker 19.03.6 ...
🚀  Launching Kubernetes ... 
🌟  Enabling addons: default-storageclass, storage-provisioner
⚠️  Enabling 'storage-provisioner' returned an error: running callbacks: [addon apply: sudo KUBECONFIG=/var/lib/minikube/kubeconfig /var/lib/minikube/binaries/v1.17.3/kubectl apply -f /etc/kubernetes/addons/storage-provisioner.yaml: Process exited with status 1
stdout:
serviceaccount/storage-provisioner created
clusterrolebinding.rbac.authorization.k8s.io/storage-provisioner created

stderr:
Error from server (ServerTimeout): error when creating "/etc/kubernetes/addons/storage-provisioner.yaml": No API token found for service account "storage-provisioner", retry after the token is automatically created and added to the service account
]
⌛  Waiting for cluster to come online ...
🏄  Done! kubectl is now configured to use "minikube"

</details>

**The output of the `minikube logs` command**:<details>

</details>

**The operating system version**:
Linux manjaro-main 5.5.2-1-MANJARO #1 SMP PREEMPT Tue Feb 4 19:22:35 UTC 2020 x86_64 GNU/Linux

**Comment by medyagh** [comment]
@jwuman  I belive that is a warrning for an addon, it might actually just work fine but warn about it could u please verify if there is anything broken other than warrnning ?

**Comment by amanhigh** [comment]
I am getting same error and once this warning appears minikube dashboard doesn't start.

For me
`minikube starts` working fine. I have no extra config.

When i try to use Dynamic Tokens using I get same error.
minikube start --extra-config=apiserver.service-account-issuer=api --extra-config=apiserver.service-account-signing-key-file=/var/lib/minikube/certs/apiserver.key --extra-config=apiserver.service-account-api-audiences=api

<details>🔥  Deleting "minikube" in virtualbox ...
💀  Removed all traces of the "minikube" cluster.
😄  minikube v1.7.3 on Darwin 10.14.5
✨  Automatically selected the virtualbox driver
🔥  Creating virtualbox VM (CPUs=2, Memory=2000MB, Disk=6000MB) ...
🐳  Preparing Kubernetes v1.17.3 on Docker 19.03.6 ...
    ▪ apiserver.service-account-issuer=api
    ▪ apiserver.service-account-signing-key-file=/var/lib/minikube/certs/apiserver.key
    ▪ apiserver.service-account-api-audiences=api
🚀  Launching Kubernetes ...
🌟  Enabling addons: default-storageclass, storage-provisioner
⚠️  Enabling 'storage-provisioner' returned an error: running callbacks: [addon apply: sudo KUBECONFIG=/var/lib/minikube/kubeconfig /var/lib/minikube/binaries/v1.17.3/kubectl apply -f /etc/kubernetes/addons/storage-provisioner.yaml: Process exited with status 1
stdout:
serviceaccount/storage-provisioner created
clusterrolebinding.rbac.authorization.k8s.io/storage-provisioner created

stderr:
Error from server (ServerTimeout): error when creating "/etc/kubernetes/addons/storage-provisioner.yaml": No API token found for service account "storage-provisioner", retry after the token is automatically created and added to the service account
]
⌛  Waiting for cluster to come online ...
🏄  Done! kubectl is now configured to use "minikube"
</details>

Event after waiting for few minutes (normally it starts in a minute).

<details>
🔌  Enabling dashboard ...
🤔  Verifying dashboard health ...
🚀  Launching proxy ...
🤔  Verifying proxy health ...
💣  http://127.0.0.1:56947/api/v1/namespaces/kubernetes-dashboard/services/http:kubernetes-dashboard:/proxy/ is not accessible: Temporary Error: unexpected response code: 404
</details>

**Comment by amanhigh** [comment]
> apiserver.service-account-signing-ke

For me this got solved as i was providing wrong Public Private Keys.

Adding Flags `--extra-config="apiserver.service-account-key-file=/var/lib/minikube/certs/sa.pub" --extra-config="apiserver.service-account-signing-key-file=/var/lib/minikube/certs/sa.key"` Solved the issue.

New Working Command
`minikube start --extra-config="apiserver.service-account-api-audiences=api" --extra-config="apiserver.service-account-issuer=api" --extra-config="apiserver.service-account-key-file=/var/lib/minikube/certs/sa.pub" --extra-config="apiserver.service-account-signing-key-file=/var/lib/minikube/certs/sa.key"`

**Comment by tstromberg** [comment]
I'm closing this issue as it hasn't seen activity in awhile, and it's unclear if this issue still exists. If this issue does continue to exist in the most recent release of minikube, please feel free to re-open it. 

Thank you for opening the issue!

**Comment by hbel** [comment]
I can confirm that the issue still persists. running minikube 1.11.0 on a Ubuntu 20.04 WSL2 with

```minikube start``` on WSL2 with Docker Desktop installed leads to the following error:

```
😄  minikube v1.11.0 on Ubuntu 20.04
✨  Automatically selected the docker driver
👍  Starting control plane node minikube in cluster minikube
💾  Downloading Kubernetes v1.18.3 preload ...
    > preloaded-images-k8s-v3-v1.18.3-docker-overlay2-amd64.tar.lz4: 526.01 MiB^[[1;7C
🔥  Creating docker container (CPUs=2, Memory=3100MB) ...
🐳  Preparing Kubernetes v1.18.3 on Docker 19.03.2 ...
    ▪ kubeadm.pod-network-cidr=10.244.0.0/16
🔎  Verifying Kubernetes components...
❗  Enabling 'default-storageclass' returned an error: running callbacks: [Error making standard the default storage class: Error listing StorageClasses: Get https://172.17.0.3:8443/apis/storage.k8s.io/v1/storageclasses: dial tcp 172.17.0.3:8443: i/o timeout]
🌟  Enabled addons: default-storageclass, storage-provisioner
```

**Comment by filipe-costa** [comment]
> I can confirm that the issue still persists. running minikube 1.11.0 on a Ubuntu 20.04 WSL2 with
> 
> `minikube start` on WSL2 with Docker Desktop installed leads to the following error:
> 
> ```
> 😄  minikube v1.11.0 on Ubuntu 20.04
> ✨  Automatically selected the docker driver
> 👍  Starting control plane node minikube in cluster minikube
> 💾  Downloading Kubernetes v1.18.3 preload ...
>     > preloaded-images-k8s-v3-v1.18.3-docker-overlay2-amd64.tar.lz4: 526.01 MiB^[[1;7C
> 🔥  Creating docker container (CPUs=2, Memory=3100MB) ...
> 🐳  Preparing Kubernetes v1.18.3 on Docker 19.03.2 ...
>     ▪ kubeadm.pod-network-cidr=10.244.0.0/16
> 🔎  Verifying Kubernetes components...
> ❗  Enabling 'default-storageclass' returned an error: running callbacks: [Error making standard the default storage class: Error listing StorageClasses: Get https://172.17.0.3:8443/apis/storage.k8s.io/v1/storageclasses: dial tcp 172.17.0.3:8443: i/o timeout]
> 🌟  Enabled addons: default-storageclass, storage-provisioner
> ```

I also have this issue, and it feels like it hangs and doesn't do anything else.

**Comment by BlacksBart** [comment]
I confirm same.  minikube v1.11.0 Ubuntu 20.04 WSL2
minikube start
😄  minikube v1.11.0 on Ubuntu 20.04
✨  Automatically selected the docker driver
👍  Starting control plane node minikube in cluster minikube
🔥  Creating docker container (CPUs=2, Memory=2300MB) ...
🐳  Preparing Kubernetes v1.18.3 on Docker 19.03.2 ...
    ▪ kubeadm.pod-network-cidr=10.244.0.0/16
🔎  Verifying Kubernetes components...
❗  Enabling 'default-storageclass' returned an error: running callbacks: [Error making standard the default storage class: Error listing StorageClasses: Get https://172.17.0.3:8443/apis/storage.k8s.io/v1/storageclasses: dial tcp 172.17.0.3:8443: i/o timeout]
🌟  Enabled addons: default-storageclass, storage-provisioner

**Comment by verganas** [comment]
I was getting the same issue and go it fixed(workaround?) by deleting ~/.kube/config and resetting my KUBECONFIG var. Maybe this helps.

**Comment by kleytonohare** [comment]
And I can also confirm it by running minikube v1.11.0 on a Debian 10.4 WSL2 with `minikube start`

😄  minikube v1.11.0 on Debian 10.4
✨  Automatically selected the docker driver
👍  Starting control plane node minikube in cluster minikube
🔥  Creating docker container (CPUs=2, Memory=3100MB) ...
🐳  Preparing Kubernetes v1.18.3 on Docker 19.03.2 ...
    ▪ kubeadm.pod-network-cidr=10.244.0.0/16
🔎  Verifying Kubernetes components...
❗  Enabling 'default-storageclass' returned an error: running callbacks: [Error making standard the default storage class: Error listing StorageClasses: Get https://172.17.0.3:8443/apis/storage.k8s.io/v1/storageclasses: dial tcp 172.17.0.3:8443: i/o timeout]
🌟  Enabled addons: default-storageclass, storage-provisioner

💣  failed to start node: startup failed: Wait failed: node pressure: list nodes: Get https://172.17.0.3:8443/api/v1/nodes: dial tcp 172.17.0.3:8443: i/o timeout

😿  minikube is exiting due to an error. If the above message is not useful, open an issue:
👉  https://github.com/kubernetes/minikube/issues/new/choose



I don't know what else to do.

**Comment by mikemountjoy** [comment]
I was able to get **minikube (v1.12.0-beta.0)**  running on **Ubuntu 20.04 under wsl2**

Running: 
**Windows 2004 (Build 20161.1000)**

**Docker Desktop 2.3.0.3 (with "Use the WSL 2 based engine", 
"Expose daemon on tcp://localhost:2375 without TLS",
"Enable integration with additional distros " ticked)**

I installed **minikube v1.12.0-beta.0**  in Ubuntu 20.04

```
curl -Lo minikube https://github.com/kubernetes/minikube/releases/download/v1.12.0-beta.0/minikube-linux-amd64 && sudo chmod +x minikube && sudo mv minikube /usr/local/bin/


sudo mkdir /sys/fs/cgroup/systemd
sudo mount -t cgroup -o none,name=systemd cgroup /sys/fs/cgroup/systemd

export MINIKUBE_PROFILE=localdev
minikube delete -p ${MINIKUBE_PROFILE}
minikube start -p ${MINIKUBE_PROFILE} --cpus 2 --memory 12288 \
               --kubernetes-version=v1.17.5 \
               --vm-driver=docker \
               --disk-size=30g \
               --extra-config=apiserver.enable-admission-plugins="LimitRanger,NamespaceExists,NamespaceLifecycle,ResourceQuota,ServiceAccount,DefaultStorageClass,MutatingAdmissionWebhook"
minikube status -p ${MINIKUBE_PROFILE}
```

Output:
```
😄  [localdev] minikube v1.12.0-beta.0 on Ubuntu 20.04
    ▪ MINIKUBE_PROFILE=localdev
✨  Using the docker driver based on user configuration
🆕  Kubernetes 1.18.3 is now available. If you would like to upgrade, specify: --kubernetes-version=v1.18.3
👍  Starting control plane node localdev in cluster localdev
🔥  Creating docker container (CPUs=2, Memory=12288MB) ...
🐳  Preparing Kubernetes v1.17.5 on Docker 19.03.2 ...
    ▪ apiserver.enable-admission-plugins=LimitRanger,NamespaceExists,NamespaceLifecycle,ResourceQuota,ServiceAccount,DefaultStorageClass,MutatingAdmissionWebhook
    ▪ kubeadm.pod-network-cidr=10.244.0.0/16
🔎  Verifying Kubernetes components...
🌟  Enabled addons: default-storageclass, storage-provisioner
🏄  Done! kubectl is now configured to use "localdev"

type: Control Plane
host: Running
kubelet: Running
apiserver: Running
kubeconfig: Configured
```

I then ran **Skaffold** and accessed the sample app from Window 10

```
echo "Download and Install Skaffold"
sudo rm -Rf /usr/local/bin/skaffold
sudo rm -Rf skaffold
curl -Lo skaffold https://storage.googleapis.com/skaffold/releases/latest/skaffold-linux-amd64
chmod +x skaffold
sudo mv skaffold /usr/local/bin

echo "Navigate to nodejs example"
git clone https://github.com/GoogleContainerTools/skaffold.git
cd skaffold/examples/nodejs

skaffold config set --kube-context ${MINIKUBE_PROFILE} local-cluster true
skaffold dev --status-check=false
```

Output:
```
[node-689f77dbd5-fttdk node] > backend@1.0.0 development /home/node/app
[node-689f77dbd5-fttdk node] > nodemon src/index.js
[node-689f77dbd5-fttdk node]
[node-689f77dbd5-fttdk node] [nodemon] 2.0.3
[node-689f77dbd5-fttdk node] [nodemon] to restart at any time, enter `rs`
[node-689f77dbd5-fttdk node] [nodemon] watching path(s): *.*
[node-689f77dbd5-fttdk node] [nodemon] watching extensions: js,mjs,json
[node-689f77dbd5-fttdk node] [nodemon] starting `node src/index.js`
[node-689f77dbd5-fttdk node] Example app listening on port 3000!
```
![image](https://user-images.githubusercontent.com/1186160/87038719-10664c80-c1e6-11ea-9522-9795021075fb.png)

I then installed istio and Knative and I was able to navigate to Kiali to view status from Windows 10

![image](https://user-images.githubusercontent.com/1186160/87038579-dac16380-c1e5-11ea-9119-a3cd783a397e.png)

**Comment by filipe-costa** [comment]
I think if it would be possible to run minikube without disabling TLS, that would be the best approach.

**Comment by mikemountjoy** [comment]
> I think if it would be possible to run minikube without disabling TLS, that would be the best approach.

Agreed. I have tested my setup and you are correct. I have edited my post above accordingly.

**Comment by kleytonohare** [comment]
> 
> 
> I was able to get **minikube (v1.12.0-beta.0)** running on **Ubuntu 20.04 under wsl2**
> 
> Running:
> **Windows 2004 (Build 20161.1000)**
> 
> **Docker Desktop 2.3.0.3 (with "Use the WSL 2 based engine", "Enable integration with additional distros " ticked)**
> 
> **%USERPROFILE%/.wslconfig**
> 
> ```
> [wsl2]
> debugConsole=true
> pageReporting=true
> nestedVirtualization=true
> memory=12GB # Limits VM memory in WSL 2 to 12 GB
> processors=4 # Makes the WSL 2 VM use four virtual processors
> ```
> 
> I installed **minikube v1.12.0-beta.0** in Ubuntu 20.04
> 
> ```
> curl -Lo minikube https://github.com/kubernetes/minikube/releases/download/v1.12.0-beta.0/minikube-linux-amd64 && sudo chmod +x minikube && sudo mv minikube /usr/local/bin/
> 
> 
> sudo mkdir /sys/fs/cgroup/systemd
> sudo mount -t cgroup -o none,name=systemd cgroup /sys/fs/cgroup/systemd
> 
> export MINIKUBE_PROFILE=localdev
> minikube delete -p ${MINIKUBE_PROFILE}
> minikube start -p ${MINIKUBE_PROFILE} --cpus 2 --memory 12288 \
>                --kubernetes-version=v1.17.5 \
>                --vm-driver=docker \
>                --disk-size=30g \
>                --extra-config=apiserver.enable-admission-plugins="LimitRanger,NamespaceExists,NamespaceLifecycle,ResourceQuota,ServiceAccount,DefaultStorageClass,MutatingAdmissionWebhook"
> minikube status -p ${MINIKUBE_PROFILE}
> ```
> 
> Output:
> 
> ```
> 😄  [localdev] minikube v1.12.0-beta.0 on Ubuntu 20.04
>     ▪ MINIKUBE_PROFILE=localdev
> ✨  Using the docker driver based on user configuration
> 🆕  Kubernetes 1.18.3 is now available. If you would like to upgrade, specify: --kubernetes-version=v1.18.3
> 👍  Starting control plane node localdev in cluster localdev
> 🔥  Creating docker container (CPUs=2, Memory=12288MB) ...
> 🐳  Preparing Kubernetes v1.17.5 on Docker 19.03.2 ...
>     ▪ apiserver.enable-admission-plugins=LimitRanger,NamespaceExists,NamespaceLifecycle,ResourceQuota,ServiceAccount,DefaultStorageClass,MutatingAdmissionWebhook
>     ▪ kubeadm.pod-network-cidr=10.244.0.0/16
> 🔎  Verifying Kubernetes components...
> 🌟  Enabled addons: default-storageclass, storage-provisioner
> 🏄  Done! kubectl is now configured to use "localdev"
> 
> type: Control Plane
> host: Running
> kubelet: Running
> apiserver: Running
> kubeconfig: Configured
> ```



OMG... thank you soooo much. This is unbelievable

**Comment by mikemountjoy** [comment]
Also tested as working with latest production release **1.12.0** running on Ubuntu 20.04 under wsl2

**Comment by ozbillwang** [comment]
got the same issue on MacOS

```
$ minikube start
😄  minikube v1.12.1 on Darwin 10.15.6
✨  Automatically selected the docker driver. Other choices: hyperkit, virtualbox
👍  Starting control plane node minikube in cluster minikube
🔥  Creating docker container (CPUs=2, Memory=4000MB) ...
🐳  Preparing Kubernetes v1.18.3 on Docker 19.03.2 ...
🔎  Verifying Kubernetes components...
❗  Enabling 'default-storageclass' returned an error: running callbacks: [Error making standard the default storage class: Error listing StorageClasses: Unauthorized]
🌟  Enabled addons: default-storageclass, storage-provisioner
```

**Comment by ashmckenzie** [comment]
Deleting both `~/.kube` and `~/.minikube` directories on macOS solved this for me.

**Comment by mikemountjoy** [comment]
I ran a quick test using docker on my own machine and minikube started successfully.

Personally I use **hyperkit** (installed as part of Docker Desktop for Mac ) rather than **docker** as my --vm-driver

```
export MINIKUBE_PROFILE=localdev
minikube delete -p ${MINIKUBE_PROFILE}
minikube start  -p ${MINIKUBE_PROFILE} --cpus 2 --memory 4000 \
               --vm-driver=docker \
               --disk-size=30g \
               --extra-config=apiserver.enable-admission-plugins="LimitRanger,NamespaceExists,NamespaceLifecycle,ResourceQuota,ServiceAccount,DefaultStorageClass,MutatingAdmissionWebhook"
minikube status -p ${MINIKUBE_PROFILE}
```

```
😄  [localdev] minikube v1.12.1 on Darwin 10.15.6
    ▪ MINIKUBE_PROFILE=localdev
✨  Using the docker driver based on user configuration
👍  Starting control plane node localdev in cluster localdev
🚜  Pulling base image ...
💾  Downloading Kubernetes v1.18.3 preload ...
    > preloaded-images-k8s-v4-v1.18.3-docker-overlay2-amd64.tar.lz4: 526.27 MiB
🔥  Creating docker container (CPUs=2, Memory=4000MB) ...
🐳  Preparing Kubernetes v1.18.3 on Docker 19.03.2 ...
    ▪ apiserver.enable-admission-plugins=LimitRanger,NamespaceExists,NamespaceLifecycle,ResourceQuota,ServiceAccount,DefaultStorageClass,MutatingAdmissionWebhook
🔎  Verifying Kubernetes components...
🌟  Enabled addons: default-storageclass, storage-provisioner
🏄  Done! kubectl is now configured to use "localdev"

localdev
type: Control Plane
host: Running
kubelet: Running
apiserver: Running
kubeconfig: Configured
```

**Comment by ozbillwang** [comment]
@ashmckenzie 

Thanks a lot. Clean the folder `~/.minikube` is the key to fix this issue. No need clean `~/.kube`

```
$ minikube start
😄  minikube v1.12.1 on Darwin 10.15.6
✨  Automatically selected the docker driver. Other choices: hyperkit, virtualbox
👍  Starting control plane node minikube in cluster minikube
💾  Downloading Kubernetes v1.18.3 preload ...
    > preloaded-images-k8s-v4-v1.18.3-docker-overlay2-amd64.tar.lz4: 526.27 MiB^@
^@🔥  Creating docker container (CPUs=2, Memory=4000MB) ...
🐳  Preparing Kubernetes v1.18.3 on Docker 19.03.2 ...
🔎  Verifying Kubernetes components...
🌟  Enabled addons: default-storageclass, storage-provisioner
🏄  Done! kubectl is now configured to use "minikube"
```

**Comment by allir** [comment]
Thanks @ashmckenzie @ozbillwang 

I had the same issue on macOS 1.15.6 and miniukbe v1.12.3. Deleting the `~/.minikube` folder solved it for me as well!

Not sure if it's related but I hadn't used minikube in a long while so I also ran into other issues first where `brew install minikube` wouldn't link the binary, I still had an older "cask" version of minikube installed which I had to `brew cask remove minikube` to get rid of and than re-install. Guessing the old config left in `~/.minikube` was maybe incompatible between versions?
